### DISCUSSION TITLE
Quick Launch Challenges

### DISCUSSION THREAD

**Discussion by tcorbet** [valinet/ExplorerPatcher#4062] [Q&A]
I love and use ExplorerPatcher (EP) for a wide range of User Experience (UE) features that you enable and are crucial to my productivity.  Probably the most important is the ability to manage a large number of Quick Launch icons taking up minimal screen space because I use small icons with no text or title.  Over the many updates to EP that I have had in 5+ years of use, mostly things have gone quite well including my finally being able to get the latest version working with Windows 11 24H2.

I have, however, encountered a great deal of difficulty in maintaining the position -- ordering more accurately -- of those multiple icons.  Using the UE -- which I presume to be from the original Windows code, not something that EP has modified -- it is somewhat challenging just to relocate/reorder the icons, but eventually I am able to re-establish the desired layout.  The problem is that I have no fool-proof set of procedures/gestures to assure that the final arrangement of the Quick Launch icons will be preserved from login session to login session.

What should be the procedure/gesture to save the layout of the Quick Launch section of the Task Bar?  [I trust that it is clear that this is not a question about the toggling of the "Lock the taskbar" option.  I understand that toggle and do occasionally use it to re-orient the position of the Quick Launch section of the Task Bar, but the matter of maintaining the ordering within that section is not dependent upon the "Lock the taskbar" toggle.]

**Comment by pyrates999** [comment]
https://github.com/valinet/ExplorerPatcher/discussions/3608#discussioncomment-10492461

https://github.com/valinet/ExplorerPatcher/discussions/3774#discussion-7289141

**Comment by tcorbet** [comment]
![Desktop](https://github.com/user-attachments/assets/3fa36b39-5f3f-49fa-8e83-0f47f53a9ab3)

Thank you, pyrates999, for providing links to two other Discussion Topics which relate to the installation, configuration and use of a Quick Launch toolbar inside an ExplorerPatcher (EP) Windows 10-like Toolbar.  I apologize for not providing a helpful screen shot to accompany my topic #4062.  It likely would have helped anyone attempting to apprehend my question(s) and suggest ideas and solutions to the problem.  

Unlike the initial challenges that the owners of topics #3608 and #3774 -- even including the expanded discussions which evolved from those two initial challenges -- my challenge does not involve either any difficulty in knowing how to have the EP-10 Taskbar provide a Quick Launch set of icon/links nor in positioning that taskbar within the main one.  My challenge was only in learning what sort of UI event must be used by whomever's code -- the original from MS or any aspect of your modifications thereto -- maintains state for the visible elements of a Quick Launch toolbar in the face of layout gestures that are used to position the individual icon/links within the x-y coordinate space available.  As I indicated in my original posting, after performing the steps to arrange the set of icons in the preferred order, I was unable to get persistence in the final layout between login sessions.

I am mostly taking the time to more thoroughly document and discuss this sort of a challenge in the hopes that some other user having any similar challenge will one day stumble upon this text and it may be of assistance.  I have now discovered what I believe the procedure/process must be in order to coerce the taskbar layout code into performing a save of its state.  Clearly a move/relocate of the icons does not coerce a state save.  So, I have now chosen this work-around solution:

A.  Use whatever set of drag-drop gestures will result in the desired layout.
B.  Drag a new link and drop it into the Quick Launch screen geography.
C.  Right-click the newly introduced icon to delete it.

Those add/delete events will cause the responsible code to save the state such that after a logout/login the layout of the suite of icons within the Quick Launch area of the main Taskbar will be preserved.
